src/manifest.loader.ts

```diff
@@ -1,16 +1,50 @@
 import { fsa } from '@linzjs/s3fs';
-import { promises as fs } from 'fs';
+import * as fs from 'fs';
+import { join } from 'path';
 import { logger } from './log';
 import { Manifest, ManifestFile } from './manifest';
 
 const BackupExtension = '.1';
+export const MANIFEST_FILE_NAME = 'manifest.json';
 
 function isManifestPath(fileName: string): boolean {
   if (fileName.endsWith('.json' + BackupExtension)) return true;
   if (fileName.endsWith('.json')) return true;
   return false;
 }
 
+/**
+ * Return true if manifest dataPath are different
+ *
+ * @param manifestA
+ * @param manifestB
+ * @returns
+ */
+export function isManifestsDifferent(manifestA: ManifestLoader, manifestB: ManifestLoader): boolean {
+  if (manifestA.dataPath !== manifestB.dataPath) return true;
+  return false;
+}
+
+/**
+ * Return true if a manifest.json generated for the same files exists in the destination folder.
+ *
+ * @param currentManifest
+ * @param destPath
+ * @returns
+ */
+export async function isDifferentManifestExists(currentManifest: ManifestLoader, destPath: string): Promise<boolean> {
+  if (!destPath.endsWith('/')) destPath = destPath + '/';
+  const existingManifestPath = join(destPath, MANIFEST_FILE_NAME);
```

Comment:
we cannot use `path.join` with uris such as  `s3://foo/bar` as it breaks the URI use `fsa.join`